### DISCUSSION TITLE
Issue with function "annotate", when annotating a pole figure within a GUI

### DISCUSSION THREAD

**Discussion by RellieGoddard** [mtex-toolbox/mtex#2102] [Ask Anything]
I'm currently working on a GUI to simplify the application of a method my co-authors and I have developed. As part of the process I identify the n most common maxima in the c-axes of a collection of grains: 

```
app.SF_1pg_0001 = calcDensity(app.vec0001,'halfwidth',app.CallingApp.HalfwidthEditField.Value*degree, 'Weights', app.CallingApp.EBSDdata_qz_Ori1p_weights);
[~,app.modes_0001] = max(app.SF_1pg_0001,'numLocal',app.NooforientationsEditField.Value);
```
Where  _app.SF_1pg_001_ is a spherical density function of the c-axis orientations (_app.vec0001_), and _app.NooforientationsEditField.Value,_ is the number of maxima you want to select. I want to then plot the points using _"annotate"_ on a pole figure, which is allocated/plotted in _app.UIAxes_. The code, which works when coding in Matlab normally, is as follows:
```
for a = 1:1:app.NooforientationsEditField.Value;
    hold(app.UIAxes);
    annotate(app.modes_0001(a),'antipodal','label',num2str(a),'backgroundcolor','w', 'parent', app.UIAxes);
end
```
However, in the app I get the following error after only the first point is plotted: 

> Current plot held
> Undefined function 'drawNow' for input arguments of
> type 'double'.
> Error in annotate (line 7)
> drawNow(gcm);

When I explore the issue it appears that _gcm_ is empty when trying to use annotate in the app, but contains the figure information when plotting in Matlab normally:

> In app:
> gcm =
> []

When using Matlab normally:

>  gcm
> ans = 
> mtexFigure with properties:
> parent: [1×1 Figure]
> children: [1×1 Axes]
> cBarAxis: []
> innerPlotSpacing: 21
> keepAspectRatio: 1
> nrows: 1
> ncols: 1
> axisWidth: 448
> axisHeight: 448
> cbx: 0
> cby: 0
> tightInset: [8.5122e-06 1.5846 8.5122e-06 28.3600]
> figTightInset: [10 10 10 10]
> layoutMode: 'auto'
> figSizeFactor: 0
> currentAxes: [1×1 Axes]
> currentId: 1
> axesWidth: 468.0000
> axesHeight: 497.9446
> outerPlotSpacing: 10
> dataCursorMenu: [1×1 ContextMenu]

 I've tried to set _gcm_ to be _app.UIAxes_, but that doesn't fix the problem, and the data in _gcm_ appears very differently, like so:

> gcm = 
> UIAxes (pdf) with properties:
> XLim: [-1.4708 1.4708]
> YLim: [-1.4708 1.4708]
> XScale: 'linear'
> YScale: 'linear'
> GridLineStyle: '-'
> Position: [53 119 434 380]
> Units: 'pixels'

Please let me know if anything needs clarification. I should mention the data I am trying to plot on the pole figure are vectors, with x/y/z co-ordinates, hence plotPDF does not work. 

I would appreciate any advice the community has, as I don't see many options in the help section of MTEX. If there are other, better, functions that annotate I'm happy to change!

**Comment by ralfHielscher** [comment]
You may want to change the function `annotate.m` as follows
```matlab
function annotate(obj,varargin)
% annotate to a existing figure

aS = getMTEXpref('annotationStyle');
plot(obj,aS{:},'add2all',varargin{:});
pause(0.01);

if ~isempty(gcm), drawNow(gcm); end

end
```
